ksql-engine/src/main/java/io/confluent/ksql/query/id/QueryIdGenerator.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.query.id;
+
+/**
+ * Generator used to provide query ids
+ */
+public interface QueryIdGenerator {
+
+  String getNext();
+
+  long peekNext();
```

Comment:
do we need this method? I only see its usage in tests and I think it causes this to be a leaky abstraction

Reply:
It's used to initialize the sandbox in HybridQueryIdGenerator by grabbing what would've been the next id from the current active generator.

Reply:
ah I missed that! I don't think we should be doing that though - we should be able to create a sandbox from both of the delegates and wrap them in a new hybrid one. that way we don't break abstraction barriers

Reply:
this function is being removed now after updates to how sandbox generators are made